### PR TITLE
doc: Introduce reqparse, the plain text requirement parser

### PR DESCRIPTION
**Pull request by robertbaldyga**
Signed-off-by: Robert Baldyga <robert.baldyga@intel.com>

### CODE REVIEW

File: doc/reqparse [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    if state == State.REQ_TEXT:
+            reqs.append(current_entity)
+    return (group, reqs)
+
+
+parser = argparse.ArgumentParser(prog="reqparse")
+parser.add_argument(
+    "-o", "--output", action="store",
+    help="Output file"
+)
```
Comment (Ostrokrzew) [inline doc/reqparse:126]: I know the user should explicitly specify output file, but what do You think about adding suffix dependent on output format?

**Review by Ostrokrzew — CHANGES_REQUESTED** [review]

File: doc/reqparse [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    "-f", "--format", action="store",
+    choices=["html", "markdown"], default="markdown",
+    help="Output format {html|markdown}",
+)
+
+args = parser.parse_args(sys.argv[1:])
+
+buf = ""
+with io.StringIO() as output:
+    for filename in glob.glob("requirements/*"):
```
[the remark this answers is not in the document]
Comment (robertbaldyga) [reply doc/reqparse:137]: That was actually the plan.

File: doc/reqparse (hunk 1 above)
Comment (robertbaldyga) [reply doc/reqparse:126]: This is not Linux-style behavior.

**Review by rafalste — APPROVED** [review]

**Review by Ostrokrzew — APPROVED** [review]

File: doc/reqparse [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        group, reqs = parse_file(filename)
+        output.write(f"# {group.header['group']}\n")
+        output.write(group.text)
+        for r in reqs:
+            output.write(f"## {r.header['title']}\n")
```
Comment (mmichal10) [inline doc/reqparse:142]: `###` instead of `##` would increase html's readability

File: doc/reqparse [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+parser.add_argument(
+    "-f", "--format", action="store",
+    choices=["html", "markdown"], default="markdown",
+    help="Output format {html|markdown}",
+)
+
+args = parser.parse_args(sys.argv[1:])
+
+buf = ""
+with io.StringIO() as output:
```
Comment (mmichal10) [inline doc/reqparse:136]: How about adding a table of contents? It only requires initializing the `output` with `output.write("[TOC]\n")` and adding `toc` to list the list of extensions.

Comment (robertbaldyga) [reply doc/reqparse:136]: Let's rather make it with JS/CSS.

File: doc/reqparse (hunk 3 above)
Comment (robertbaldyga) [reply doc/reqparse:142]: It's matter of CSS formatting.

**Pull request merged**